go/vt/vtgate/evalengine/string.go

```diff
@@ -265,3 +268,37 @@ func (builtinRepeat) typeof(env *ExpressionEnv, args []Expr) (sqltypes.Type, fla
 
 	return sqltypes.VarChar, f1
 }
+
+type builtinConv struct {
+}
+
+func (builtinConv) call(env *ExpressionEnv, args []EvalResult, result *EvalResult) {
+	inarg := &args[0]
+	fromBase := args[1]
+	toBase := args[2]
+
+	if inarg.isNull() || fromBase.isNull() || toBase.isNull() || fromBase.int64() < 2 || fromBase.int64() > 36 || toBase.int64() < 2 || toBase.int64() > 36 {
+		result.setNull()
+		return
+	}
+
+	inarg.makeSignedIntegral()
```

Comment:
You don't want a signed integral for `inarg`. You want the raw textual bytes so you can process them.